### ISSUE TITLE
Error when combining implicit solvent and salt concentration (implicitSolventKappa)

### ISSUE THREAD

**Issue by FloLangenfeld** [openmm/openmm#3417]
I just met an error when I tried to launch a simulation.

Here is a minimal example to reproduce the error, using the alanine-dipeptide-implicit.pdb file from the tests:
```
from openmm.app import ForceField, PDBFile
from openmm.unit import nanometer

pdb = PDBFile('alanine-dipeptide-implicit.pdb')
forcefield = ForceField('amber14-all.xml', 'implicit/obc2.xml')

system = forcefield.createSystem(pdb.topology, implicitSolventKappa=1.0 / nanometer)
```

The error is:
```
Traceback (most recent call last):
  File "implicit_solvent_bug.py", line 7, in <module>
    system = forcefield.createSystem(pdb.topology, implicitSolventKappa=1.0 / nanometer)
  File "openmm/app/forcefield.py", line 1376, in createSystem
    exec(script, locals())
  File "<string>", line 24, in <module>
  File "openmm/app/internal/customgbforces.py", line 687, in __init__
    _createEnergyTerms(self, solventDielectric, soluteDielectric, SA, cutoff, kappa, 0.009)
  File "openmm/app/internal/customgbforces.py", line 360, in _createEnergyTerms
    params = "; solventDielectric=%.16g; soluteDielectric=%.16g; kappa=%.16g; offset=%.16g" % (solventDielectric, soluteDielectric, kappa, offset)
TypeError: Quantity.__float__ returned non-float (type Quantity)
```

If one removes the `implicitSolventKappa` parameter, no error is produced.
I changed the `kappa` variables to `kappa._value` in the `_createEnergyTerms` function of `customgbforces.py` (not only the first occurrence, but also the following ones in the if/else tests), and it solves the issue.

Best regards,

Florent

**Comment by peastman** [comment]
Thanks!  We should make it handle units correctly.

**Comment by peastman** [comment]
This was fixed in #3579.
